modules/core/include/opencv2/core/dualquaternion.inl.hpp

```diff
@@ -387,6 +392,94 @@ DualQuat<T> DualQuat<T>::sclerp(const DualQuat<T> &q0, const DualQuat<T> &q1, co
     return v0 * v0inv1.power(t, QUAT_ASSUME_UNIT);
 }
 
+template <typename T>
+DualQuat<T> DualQuat<T>::dqblend(const DualQuat<T> &q1, const DualQuat<T> &q2, const T t, QuatAssumeType assumeUnit)
+{
+    DualQuat<T> v1(q1), v2(q2);
+    if (!assumeUnit)
+    {
+        v1 = v1.normalize();
+        v2 = v2.normalize();
+    }
+    if (v1.getRotation().dot(v2.getRotation()) < 0)
+    {
+        return ((1 - t) * v1 - t * v2).normalize();
+    }
+    return ((1 - t) * v1 + t * v2).normalize();
+}
+
+template <typename T>
+DualQuat<T> DualQuat<T>::gdqblend(const std::vector<DualQuat<T>> &_dualquat, const std::vector<T> &weight, QuatAssumeType assumeUnit)
+{
+    std::vector<DualQuat<T>> dualquat(_dualquat);
+    if (!assumeUnit)
+    {
+        for (auto &dq : dualquat)
+        {
+            dq = dq.normalize();
+        }
+    }
+    size_t cn = dualquat.size();
+    if (cn == 0)
+    {
+        return DualQuat<T>(1, 0, 0, 0, 0, 0, 0, 0);
+    }
+    DualQuat<T> dq_blend = dualquat[0] * weight[0];
+    Quat<T> q0 = dualquat[0].getRotation();
```

Comment:
Here `QUAT_ASSUME_UNIT` can be used in `getRotation` calls

Reply:
OK